### ISSUE TITLE
Integration does not always load due to ?

### ISSUE THREAD

**Issue by snicker** [rsnodgrass/hass-flo-water#1]
The error below happens on `master` as well as `2.0.1`

```
2020-01-15 17:55:27 ERROR (MainThread) [homeassistant.components.switch] Error while setting up platform flo
Traceback (most recent call last):
  File "/srv/homeassistant/lib/python3.7/site-packages/homeassistant/helpers/entity_platform.py", line 150, in _async_setup_platform
    await asyncio.wait_for(asyncio.shield(task), SLOW_SETUP_MAX_WAIT)
  File "/usr/local/lib/python3.7/asyncio/tasks.py", line 442, in wait_for
    return fut.result()
  File "/usr/local/lib/python3.7/concurrent/futures/thread.py", line 57, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/homeassistant/.homeassistant/custom_components/flo/switch.py", line 43, in setup_platform
    switches.append( FloWaterValve(hass, flo, device['id']) )
  File "/home/homeassistant/.homeassistant/custom_components/flo/switch.py", line 56, in __init__
    self.update()
  File "/home/homeassistant/.homeassistant/custom_components/flo/switch.py", line 87, in update
    self._is_open = valve['target'] == 'open'
KeyError: 'target'
```

**Comment by rsnodgrass** [comment]
Thanks, I added additional logging so we can see the state of your valves *AND* changed this to just fail gracefully (no state change).  Try updating to master again and posting any warning log statements from the flo component once you restart.

**Comment by snicker** [comment]
Thanks @rsnodgrass got a chance to update this. This is the result of the logging:

```
2020-01-16 19:55:20 WARNING (SyncWorker_8) [custom_components.flo.switch] Could not update valve state for device 2df60ff8-f6f8-4cea-9910-546c9915396c: {'actionRules': [],
 'connectivity': {'rssi': -41, 'ssid': 'redacted'},
 'deviceModel': 'flo_device_075_v2',
 'deviceType': 'flo_device_v2',
 'fwProperties': {'alarm_home_high_flow_rate_shut_off_deferment': 300,
                  'alarm_home_high_water_use_shut_off_deferment': 300,
                  'alarm_home_long_flow_event_shut_off_deferment': 300,
                  'alarm_shutoff_time_epoch_sec': -1,
                  'alarm_snooze_enabled': True,
                  'alarm_suppress_duplicate_duration': 300,
                  'alarm_suppress_until_event_end': False,
                  'device_data_free_mb': 1728,
                  'device_installed': True,
                  'device_mem_available_kb': 353712,
                  'device_rootfs_free_kb': 802476,
                  'device_uptime_sec': 97832,
                  'flosense_action': 'stop',
                  'flosense_deployment_result': '',
                  'flosense_link': '',
                  'flosense_shut_off_enabled': True,
                  'flosense_shut_off_level': 2,
                  'flosense_state': 'not_available',
                  'flosense_version_app': '',
                  'flosense_version_model': '',
                  'fw_ver': '4.2.4-artifact-v2',
                  'fw_ver_a': '',
                  'fw_ver_b': '4.2.4-artifact-v2',
                  'ht_attempt_interval': 60000,
                  'ht_check_window_max_pressure_decay_limit': 0.1,
                  'ht_check_window_width': 30000,
                  'ht_max_open_closed_pressure_decay_pct_limit': 2,
                  'ht_max_pressure_growth_limit': 3,
                  'ht_max_pressure_growth_pct_limit': 3,
                  'ht_min_computable_point_limit': 3,
                  'ht_min_pressure_limit': 10,
                  'ht_min_r_squared_limit': 0.9,
                  'ht_min_slope_limit': -0.6,
                  'ht_phase_1_max_pressure_decay_limit': 6,
                  'ht_phase_1_max_pressure_decay_pct_limit': 10,
                  'ht_phase_1_time_index': 12000,
                  'ht_phase_2_max_pressure_decay_limit': 6,
                  'ht_phase_2_max_pressure_decay_pct_limit': 10,
                  'ht_phase_2_time_index': 30000,
                  'ht_phase_3_max_pressure_decay_limit': 3,
                  'ht_phase_3_max_pressure_decay_pct_limit': 5,
                  'ht_phase_3_time_index': 240000,
                  'ht_phase_4_max_pressure_decay_limit': 1.5,
                  'ht_phase_4_max_pressure_decay_pct_limit': 5,
                  'ht_phase_4_time_index': 480000,
                  'ht_pre_delay': 0,
                  'ht_recent_flow_event_cool_down': 1000,
                  'ht_retry_on_fail_interval': 900000,
                  'ht_times_per_day': 1,
                  'log_bytes_sent': 80718,
                  'log_frequency': 3600,
                  'mender_host': 'https://mender.flotech.co',
                  'motor_delay_close': 175,
                  'motor_delay_open': 0,
                  'motor_retry_count': 2,
                  'motor_timeout': 5000,
                  'pes_away_max_duration': 3600,
                  'pes_away_max_pressure': 150,
                  'pes_away_max_temperature': 226,
                  'pes_away_max_volume': 50,
                  'pes_away_min_pressure': 20,
                  'pes_away_min_temperature': 36,
                  'pes_away_v1_high_flow_rate': 8,
                  'pes_away_v1_high_flow_rate_duration': 5,
                  'pes_away_v2_high_flow_rate': 0.5,
                  'pes_away_v2_high_flow_rate_duration': 5,
                  'pes_home_high_flow_rate': 8,
                  'pes_home_high_flow_rate_duration': 20,
                  'pes_home_max_duration': 3600,
                  'pes_home_max_pressure': 150,
                  'pes_home_max_temperature': 226,
                  'pes_home_max_volume': 50,
                  'pes_home_min_pressure': 20,
                  'pes_home_min_temperature': 36,
                  'pes_moderately_high_pressure': 80,
                  'pes_moderately_high_pressure_count': 43200,
                  'pes_moderately_high_pressure_delay': 300,
                  'pes_moderately_high_pressure_period': 10,
                  'player_action': 'disabled',
                  'player_flow': 0,
                  'player_min_pressure': 40,
                  'player_pressure': 60,
                  'player_temperature': 50,
                  'power_downtime_last_24h': 85277,
                  'power_downtime_last_7days': 95851,
                  'power_downtime_last_reboot': 95851,
                  'reboot_count': 2,
                  'reboot_count_7days': 2,
                  'reboot_reason': 'power_cycle',
                  's3_bucket_host': 'api-bulk.meetflo.com',
                  'serial_number': '318418680077',
                  'system_mode': 5,
                  'telemetry_batched_enabled': True,
                  'telemetry_batched_interval': 300,
                  'telemetry_batched_pending_storage': 30,
                  'telemetry_batched_sent_storage': 30,
                  'telemetry_flow_rate': 0.20071736158298117,
                  'telemetry_pressure': 83.10931004749986,
                  'telemetry_realtime_change_gpm': 0,
                  'telemetry_realtime_change_psi': 0,
                  'telemetry_realtime_interval': 1,
                  'telemetry_realtime_session_last_epoch': 0,
                  'telemetry_realtime_sessions_7days': 0,
                  'telemetry_realtime_storage': 7,
                  'telemetry_realtime_timeout': 299,
                  'telemetry_temperature': 56.00000046340355,
                  'valve_actuation_count': 8,
                  'valve_actuation_timeout_count': 0,
                  'valve_state': 1,
                  'wifi_bytes_received': 718042,
                  'wifi_bytes_sent': 2057745,
                  'wifi_disconnections': 1,
                  'wifi_rssi': -40,
                  'wifi_sta_enc': 'psk2',
                  'wifi_sta_ssid': 'redacted',
                  'zit_auto_count': 3,
                  'zit_manual_count': 0},
 'fwVersion': '4.2.4-artifact-v2',
 'hardwareThresholds': {'gpm': {'maxValue': 35,
                                'minValue': 0,
                                'okMax': 29,
                                'okMin': 0},
                        'kPa': {'maxValue': 700,
                                'minValue': 0,
                                'okMax': 550,
                                'okMin': 210},
                        'lpm': {'maxValue': 130,
                                'minValue': 0,
                                'okMax': 110,
                                'okMin': 0},
                        'psi': {'maxValue': 100,
                                'minValue': 0,
                                'okMax': 80,
                                'okMin': 30},
                        'tempC': {'maxValue': 40,
                                  'minValue': 0,
                                  'okMax': 30,
                                  'okMin': 10},
                        'tempF': {'maxValue': 100,
                                  'minValue': 0,
                                  'okMax': 80,
                                  'okMin': 50}},
 'id': '2df60ff8-f6f8-4cea-9910-546c9915396c',
 'installStatus': {'installDate': '2020-01-12T23:34:42.829Z',
                   'isInstalled': True},
 'irrigationType': 'not_plumbed',
 'isConnected': True,
 'isPaired': True,
 'lastHeardFromTime': '2020-01-16T19:42:22Z',
 'location': {'id': '396ccb73-d319-4496-9d08-0dc1415202d2'},
 'macAddress': '3881d7375c90',
 'nickname': 'redacted',
 'notifications': {'pending': {'alarmCount': [{'count': 1,
                                               'id': 57,
                                               'severity': 'warning'}],
                               'criticalCount': 0,
                               'infoCount': 0,
                               'warningCount': 1}},
 'prvInstallation': 'none',
 'serialNumber': '318418680077',
 'shutoff': {'scheduledAt': '1970-01-01T00:00:00.000Z'},
 'systemMode': {'isLocked': True,
                'lastKnown': 'sleep',
                'shouldInherit': True,
                'target': 'sleep'},
 'telemetry': {'current': {'gpm': 0,
                           'psi': 84.70780944824219,
                           'tempF': 57.09540557861328,
                           'updated': '2020-01-16T16:12:58Z'}},
 'valve': {'lastKnown': 'open'}} / {'lastKnown': 'open'}
```

**Comment by rsnodgrass** [comment]
Thank you!  I believe this should now be fixed on master (and I removed the warning log).  Is it working for you?  If so, I'll release it.

**Comment by snicker** [comment]
Looks like it's working great now... with one caveat: the sensor values do not actually update unless I have the Flo app open on my phone. I noticed that values were only showing in the history intermittently and that they weren't updating for hours at a time.. then I decided to give that at shot. As long as I leave the app open on my phone screen, HA updates values every few seconds.

Is there some sort of command that needs to be sent via the API to turn on active monitoring from the device that the python library is skipping?
